client/src/entrypoints/snippets/snippet-chooser.js

```diff
@@ -1,95 +1,26 @@
-import $ from 'jquery';
+import { Chooser } from '../../components/ChooserWidget';
 
 /* global wagtailConfig */
 
-function createSnippetChooser(id) {
-  const chooserElement = $('#' + id + '-chooser');
-  const docTitle = chooserElement.find('.title');
-  const input = $('#' + id);
-  const editLink = chooserElement.find('.edit-link');
-  const chooserBaseUrl = chooserElement.data('chooserUrl');
-
-  /*
-  Construct initial state of the chooser from the rendered (static) HTML and arguments passed to
-  createSnippetChooser. State is either null (= no document chosen) or a dict of id, string and
-  edit_link.
-
-  The result returned from the snippet chooser modal (see wagtail.snippets.views.chooser.chosen)
-  is a superset of this, and can therefore be passed directly to chooser.setState.
-  */
-  let state = null;
-  if (input.val()) {
-    state = {
-      id: input.val(),
-      edit_link: editLink.attr('href'),
-      string: docTitle.text(),
-    };
+class SnippetChooser extends Chooser {
+  titleStateKey = 'string';
+
+  getModalUrl() {
+    let urlQuery = '';
+    if (wagtailConfig.ACTIVE_CONTENT_LOCALE) {
+      // The user is editing a piece of translated content.
+      // Pass the locale along as a request parameter. If this
+      // snippet is also translatable, the results will be
+      // pre-filtered by this locale.
+      urlQuery = '?locale=' + wagtailConfig.ACTIVE_CONTENT_LOCALE;
```

Comment:
What if the chooserBaseUrl already contains `?` 

Should we be a bit smarter here when generating a url with query Params?
https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams

Reply:
Good point! I think this might be a case of "cross that bridge when we get to it" though. In the case of the snippet chooser we know that chooserBaseUrl will always be something of the form `/admin/snippets/choose/app_label/model_name/`, and anything that introduces a querystring on top of that is, by necessity, custom code. It's reasonable to assume that any custom code doing that would intend those query params to be passed on to the modal, but at that point we're also into "reasoning about hypothetical code someone might write" territory and it's probably not worth giving it further attention until it actually comes up :-)

(FWIW, I'd suspect that anyone writing custom code doing that sort of parameter passing would: A) base it on the base chooser implementation rather than the snippet chooser; B) fully expect to have to customise this bit of JS anyhow; and C) be better off passing them from the chooser widget as additional `data-` attributes rather than using a non-standard chooserBaseUrl.)

Reply:
ok makes sense - thanks for explaining.